### PR TITLE
add data getters and ploting routines

### PR DESCRIPTION
**Pull request by nicolasaunai**
Typically allows one to do stuff like 

```python
path = "phare_jobs/initialization_tests/phare_outputs_1000"
r = Run(path)

# getting data as hierarchies

B = r.GetB(0)
E = r.GetE(0)
Ni = r.GetNi(0)
Vi = r.GetVi(0)
Fp = r.GetFlux(0, "protons")
J = r.GetJ(0)

# plotting scalars

Ni.plot()
```

image

```python

# plotting components of vectors 

Vi.plot(qty="bulkVelocity_z")

``` 

image

```python

# distribution functions

parts = r.GetParticles(0, "protons")
parts.dist_plot(select=lambda particles: particles.select(Box(2.,12.5), box_type="pos"),
                axis=("x","Vz"),dual
               bins=(100,200))

``` 
image

or in full velocity space

``` python
parts = r.GetParticles(0, "protons")
parts.dist_plot(select=lambda particles: particles.select(Box(2.,2.5), box_type="pos"),
                kde=True,
                axis=("Vx","Vy"),
               bulk=True)
``` 
image

### CODE REVIEW

**Comment by nicolasaunai** [comment]
on top of #328  :-/

File: pyphare/pyphare/pharesee/particles.py [hunk 1]
```diff
@@ -16,17 +16,26 @@ def __init__(self, **kwargs):
...
+            self.dl       = np.zeros(box.dim())+0.1
 
         else:
             self.iCells  = kwargs["icells"]
             self.deltas  = kwargs["deltas"]
             self.v       = kwargs["v"]
             self.weights = kwargs["weights"]
             self.charges = kwargs["charges"]
+            self.dl      = kwargs["dl"]
+            self._x = None
```
Comment (Dekken) [inline pyphare/pyphare/pharesee/particles.py:28]: should this be indented?

**Review by Dekken — APPROVED** [review]

File: pyphare/pyphare/pharesee/particles.py (hunk 1 above)
Comment (Dekken) [reply pyphare/pyphare/pharesee/particles.py:28]: I'm not sure it matters, but I would guess it should be outside the if

Comment (nicolasaunai) [reply pyphare/pyphare/pharesee/particles.py:28]: well I don't know what dl to put to randomely generated particles taht's why I've not put any to them... but we do not use that in tests so for now I think it's ok

**Pull request merged**